test/test.js

```diff
@@ -14,290 +19,476 @@ function log () {
   console.log.apply(console, args)
 }
 
-var testfile = __dirname + '/testfile.txt'
+var testfilePath = __dirname + '/testfile.txt'
+var testfile = fs.readFileSync(__dirname + '/testfile.txt')
+
+describe('IPFS Node.js API wrapper tests', function () {
+  var apiClients = {} // a, b, c
+  var ipfsNodes = {} // a, b, c
 
-describe('ipfs node api', function () {
-  var ipfs, ipfsNode
   before(function (done) {
     this.timeout(20000)
     log('ipfs node setup')
+    var counter = 0
+
+    if (isNode) {
+      startIndependentNode(ipfsNodes, apiClients, 'a', finish)
+      startIndependentNode(ipfsNodes, apiClients, 'b', finish)
+      startIndependentNode(ipfsNodes, apiClients, 'c', finish)
```

Comment:
I suggest doing we write a script to start karma instead of using the cli, similar to [this](https://github.com/karma-runner/gulp-karma#tldr) and then we can wrap the karma start and stop into start and stop of the nodes.